libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -482,8 +482,8 @@ class MapboxNavigation(
     fun setRoutes(routes: List<DirectionsRoute>) {
         rerouteController?.interrupt()
         routeAlternativesController.interrupt()
-        routeRefreshController.restart()
         directionsSession.routes = routes
+        routeRefreshController.restart()
```

Comment:
Does changing the order here matters? 🤔 

Edited: Yeah, it seems so 👀 

https://github.com/mapbox/mapbox-navigation-android/blob/b3aca80bcdc03cca189da2ed97acecfee1cd4295/libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationTest.kt#L295-L300